### PR TITLE
Add spinning up clients via docker-compose.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
For anyone testing this, you should end up with completely real clients that you can test via doing a package install from the UI. The EL5 client won't currently work due to an issue with the El5 packages that I still need to resolve.

**Comment by ehelms** [comment]
Needs https://github.com/Katello/katello-deploy/pull/172 to pass tests

**Comment by stbenjam** [comment]
@ehelms Needs rebase

File: README.md [hunk 1]
```diff
@@ -291,3 +291,30 @@ module_test:
   box: centos6
   options: --module-prs qpid/12
 ```
+
+## Client Testing With Docker
+
+The docker/clients directory contains setup and configuration to register clients via subscription-manager using an activation key and start katello-agent. The following are pre-requisites:
+
+ * (docker)[https://docs.docker.com/installation/]
+ * (docker-compose)[https://docs.docker.com/compose/install/]
```
Comment (stbenjam) [inline README.md]: As these are readily available (or at least F22 which I used), can you also include a cut-and-paste bit?

```
yum install -y docker docker-compose
systemctl start docker
```

**Comment by stbenjam** [comment]
Very very nice, I got some clients registered quite easily!

File: README.md (hunk 1 above)
Comment (ehelms) [inline README.md]: I suppose I can, I do think that the docker installation instructions are fairly straightforward and support whatever OS you are using including mentioning which versions it works on (e.g. it has a Fedora section that talks about how you need F20 or greater).

Comment (stbenjam) [inline README.md]: I was just suggesting the yum commands in addition to the docs.  I don't want to have to go through 2 docker docs to figure out how to install it when it's just a yum install.  We don't do that elsewhere, we give the command to install EPEL not link to the Fedora wiki.

**Comment by stbenjam** [comment]
ACK from me, I'll leave it up to you to decide on the README suggestion

**Pull request merged**